### PR TITLE
feat: support model run handler with args

### PR DESCRIPTION
**Pull request by anda-ren**
## Description
Support parameter pass in to handlers
### Features
1. List is supported
2. Dataset & Context is supported
3. User could define his/her own parameter `HandlerInput` do define how to explain the options
#### code
```python
from starwhale import (
    Context,
    Dataset,
    handler,
    IntInput,
    ListInput,
    HanderInput,
    ContextInput,
    DatasetInput,
)


class MyInput(HanderInput):
    def parse(self, user_input):

        return f"MyInput {user_input}"


class X:
    def __init__(self) -> None:
        self.a = 1

    @handler()
    def f(
        self,
        x=ListInput(IntInput),
        y=2,
        mi=MyInput(),
        ds=DatasetInput(required=True),
        ctx=ContextInput(),
    ):
        assert self.a + x[0] is 3
        assert self.a + x[1] is 2
        assert y is 2
        assert mi == "MyInput blab-la"
        assert isinstance(ds, Dataset)
        assert isinstance(ctx, Context)


@handler()
def f(
    x=ListInput(IntInput()),
    y=2,
    mi=MyInput(),
    ds=DatasetInput(required=True),
    ctx=ContextInput(),
):
    assert x[0] is 2
    assert x[1] is 1
    assert y is 2
    assert mi == "MyInput blab-la"

    assert isinstance(ds, Dataset)
    assert isinstance(ctx, Context)


```
#### bare function usage
```bash
$ swcli model run -w . -m th --handler th:f -- --x 2 --x=1 -mi=blab-la --ds mnist
✈ change current dir to /mnt/data/renyanda/code/starwhale/example/th
🚀 generate jobs yaml from modules: ('th',) , package rootdir: /mnt/data/renyanda/code/starwhale/example/th
⏳ start to run model, handler:th:f ...
🏦 runnable handlers:
            [0]: th:X.f
         *  [1]: th:f
💯 finish run, success!
```

#### function in object use case
```bash
$ swcli model run -w . -m th --handler th:X.f -- --x 2 --x=1 -mi=blab-la --ds mnist
✈ change current dir to /mnt/data/renyanda/code/starwhale/example/th
🚀 generate jobs yaml from modules: ('th',) , package rootdir: /mnt/data/renyanda/code/starwhale/example/th
⏳ start to run model, handler:th:X.f ...
🏦 runnable handlers:
         *  [0]: th:X.f
            [1]: th:f
💯 finish run, success!
```

### controller
image

## Limitations

1. no help info for these parameter options
2. positional args are not supported, only options
3. flag args are not supported

## Modules
- [ ] UI
- [ ] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [ ] run code format and lint check
- [ ] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: client/setup.py [hunk 1]
```diff
@@ -43,6 +43,7 @@
     "httpx>=0.22.0",
     "urllib3<1.27",
     "pydantic<2.0.0",  # current broker and fastapi lib code only work with pydantic < 2.0.0
+    "fire",
```
Comment (tianweidut) [inline client/setup.py:46]: `fire` and `click`(current use for command line parser) are similar libs, we maybe use only one lib. The mixed usage maybe confused.

File: client/starwhale/__init__.py [hunk 2]
```diff
@@ -58,6 +58,8 @@
     "Text",
     "Line",
     "Point",
+    "DsInput",
```
Comment (tianweidut) [inline client/starwhale/__init__.py:61]: Replace `DatasetInput` with `DsInput`?  for the more explicated usage

File: client/starwhale/api/_impl/job.py [hunk 3]
```diff
@@ -323,3 +379,11 @@ def generate_jobs_yaml(
         ),
         parents=True,
     )
+
+
+class DsInput:
```
Comment (tianweidut) [inline client/starwhale/api/_impl/job.py:384]: Are there some limitations for `DsInput` and `ContextInput`? Any object is equal to `DsInput` and `ContextInput`?

Comment (anda-ren) [reply client/starwhale/api/_impl/job.py]: done

File: client/starwhale/__init__.py (hunk 2 above)
Comment (anda-ren) [reply client/starwhale/__init__.py]: done

File: client/setup.py (hunk 1 above)
Comment (anda-ren) [reply client/setup.py]: done

**Pull request closed without merging**